dotcom-rendering/src/client/userFeatures/user-features.test.ts

```diff
@@ -46,25 +58,44 @@ const getAuthStatus = getAuthStatus_ as jest.MockedFunction<
 	typeof getAuthStatus_
 >;
 
-const setAllFeaturesData = (opts: { isExpired: boolean }) => {
-	const currentTime = new Date().getTime();
-	const msInOneDay = 24 * 60 * 60 * 1000;
-	const expiryDate = opts.isExpired
-		? new Date(currentTime - msInOneDay)
-		: new Date(currentTime + msInOneDay);
-
-	setHideSupportMessagingCookie(true);
-	setAdFreeCookie();
-	setAllowRejectAllCookie();
-	setUserFeaturesExpiryCookie(expiryDate.getTime().toString());
+const setAllBenefitsData = (opts: { isExpired: boolean }) => {
+	const daysToExpiry = opts.isExpired ? -1 : 1;
+	createOrRenewCookie(HIDE_SUPPORT_MESSAGING_COOKIE, daysToExpiry);
+	createOrRenewCookie(AD_FREE_USER_COOKIE, daysToExpiry);
+	createOrRenewCookie(ALLOW_REJECT_ALL_COOKIE, daysToExpiry);
+	createOrRenewCookie(USER_BENEFITS_EXPIRY_COOKIE, daysToExpiry);
 };
 
 beforeAll(() => {
 	window.guardian.config.page.userBenefitsApiUrl = 'fake-url';
 	window.guardian.config.tests['useUserBenefitsApiVariant'] = 'variant';
 });
 
-describe('Refreshing the features data', () => {
+const expectUserBenefitExpiryCookieHasBeenSetCorrectly = () => {
+	const expectedNextRefreshDate = new Date(
+		timeInDaysFromNow(1),
```

Comment:
Is there a danger here that the system clock could tick over to the next second between the cookie being set and running this assertion, resulting in flaky tests? Should we be freezing time in the tests?

Reply:
This check just looks at the date though not the time, so I think it's ok?